File: dailyQuestion/2020-10/10-06/python/solution_two_point.py
def sortColors(nums: [int]) -> None:
    """
    Do not return anything, modify nums in-place instead.
    """
    n = len(nums)
    p0, p2 = 0, n-1
    i = 0
    while i <= p2 :
        while i <= p2 and nums[i] == 2 :
            nums[i] , nums[p2] = nums[p2],nums[i]
            p2 -=1
        if nums[i] == 0 :
            nums[i] , nums[p0] = nums[p0] , nums[i]
            p0 += 1
        i += 1


if __name__ == "__main__" :
    nums = [2,0,2,1,1,0]
    sortColors(nums)
    print(nums)

File: dailyQuestion/2020/2020-10/10-06/golang/solution_two_point.go
package main

import (
	"fmt"
)

func main() {
	nums := []int{2, 0, 2, 1, 1, 0}
	sortColors(nums)
	fmt.Println(nums)
}

func sortColors(nums []int) {
	n := len(nums)
	p0, p1 := 0, 0
	for i := 0; i < n; i++ {
		if nums[i] == 1 {
			nums[i], nums[p1] = nums[p1], nums[i]
			p1++
		} else if nums[i] == 0 {
			nums[i], nums[p0] = nums[p0], nums[i]
			if p0 < p1 {
				nums[i], nums[p1] = nums[p1], nums[i]
			}
			p0++
			p1++
		}
	}
}
